spec/form_objects/support/allocation_form_spec.rb

```diff
@@ -1,75 +1,332 @@
-        described_class.new(allocation: updated_value, school_allocation: school_device_allocation)
+      context 'when allocation increases' do
+        let(:allocation) { 51 }
+
+        it 'return false' do
+          expect(form.save).to be_falsey
+        end
+
+        it 'do not modify any school ordering value' do
+          expect { form.save }.not_to change(school, :order_state)
+          expect { form.save }.not_to change(school, :laptop_allocation)
+          expect { form.save }.not_to change(school, :router_allocation)
+          expect { form.save }.not_to change(school, :router_cap)
+          expect { form.save }.not_to change(school, :laptop_cap)
+        end
+
+        it 'add error to school field' do
+          errors = {
+            school: ['Decreasing an allocation for a school in a virtual cap pool is currently not possible - contact the dev team to do this manually for now'],
+          }
+
+          expect(form.save).to be_falsey
+
+          expect(form.errors.messages).to eq(errors)
+        end
       end
 
-      before do
-        # Properly associate things with a virtual cap pool.
-        responsible_body.add_school_to_virtual_cap_pools!(school)
-        school_device_allocation.reload
+      context 'when allocation decreases below devices ordered' do
+        let(:allocation) { 9 }
+
+        it 'return false' do
+          expect(form.save).to be_falsey
+        end
+
+        it 'do not modify any school ordering value' do
+          expect { form.save }.not_to change(school, :order_state)
+          expect { form.save }.not_to change(school, :laptop_allocation)
+          expect { form.save }.not_to change(school, :router_allocation)
+          expect { form.save }.not_to change(school, :router_cap)
+          expect { form.save }.not_to change(school, :laptop_cap)
+        end
+
+        it 'add error to school field' do
+          errors = {
+            school: ['Allocation cannot be less than the number they have already ordered (10)'],
+          }
+
+          expect(form.save).to be_falsey
+
+          expect(form.errors.messages).to eq(errors)
+        end
       end
 
-      # Precondition
-      context 'precondition' do
-        let(:initial_value) { 1 }
-        let(:updated_value) { 2 }
+      it 'modify the school device raw allocation' do
+        expect { form.save }.to change(school, :raw_laptop_allocation).from(50).to(45)
+      end
+
+      it 'modify the pool device allocation' do
+        expect { form.save }.to change(school, :laptop_allocation).from(100).to(95)
+      end
 
-        it 'school_device_allocation#is_in_virtual_cap_pool? is true' do
-          expect(school_device_allocation.is_in_virtual_cap_pool?).to be true
+      context 'when the school cannot order' do
+        let(:order_state) { 'cannot_order' }
+        let(:requests) do
+          [
+            [
+              { 'capType' => 'DfE_RemainThresholdQty|Std_Device', 'shipTo' => '11', 'capAmount' => '40' },
```

Comment:
40 just seems like such a random number!

Reply:
Ok, fixing it